File: mobile-native/app/src/main/java/com/highloadinvest/nativeapp/ui/theme/Theme.kt
package com.highloadinvest.nativeapp.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.graphics.Color

data class AppColors(
    val canvas: Color,
    val surface: Color,
    val elevated: Color,
    val borderDefault: Color,
    val borderSubtle: Color,
    val textPrimary: Color,
    val textSecondary: Color,
    val textMuted: Color,
    val textInverse: Color,
    val accentUp: Color,
    val accentDown: Color,
    val accentBrand: Color,
    val accentWarning: Color,
    val isDark: Boolean
)

private val DarkColors = AppColors(
    canvas = DarkPalette.canvas, surface = DarkPalette.surface, elevated = DarkPalette.elevated,
    borderDefault = DarkPalette.borderDefault, borderSubtle = DarkPalette.borderSubtle,
    textPrimary = DarkPalette.textPrimary, textSecondary = DarkPalette.textSecondary,
    textMuted = DarkPalette.textMuted, textInverse = DarkPalette.textInverse,
    accentUp = DarkPalette.accentUp, accentDown = DarkPalette.accentDown,
    accentBrand = DarkPalette.accentBrand, accentWarning = DarkPalette.accentWarning,
    isDark = true
)

private val LightColors = AppColors(
    canvas = LightPalette.canvas, surface = LightPalette.surface, elevated = LightPalette.elevated,
    borderDefault = LightPalette.borderDefault, borderSubtle = LightPalette.borderSubtle,
    textPrimary = LightPalette.textPrimary, textSecondary = LightPalette.textSecondary,
    textMuted = LightPalette.textMuted, textInverse = LightPalette.textInverse,
    accentUp = LightPalette.accentUp, accentDown = LightPalette.accentDown,
    accentBrand = LightPalette.accentBrand, accentWarning = LightPalette.accentWarning,
    isDark = false
)

val LocalAppColors = staticCompositionLocalOf { DarkColors }

object AppTheme {
    val colors: AppColors
        @Composable get() = LocalAppColors.current
}

enum class ThemeMode { System, Dark, Light }

@Composable
fun HighLoadTheme(mode: ThemeMode = ThemeMode.System, content: @Composable () -> Unit) {
    val sysDark = isSystemInDarkTheme()
    val isDark = when (mode) {
        ThemeMode.System -> sysDark
        ThemeMode.Dark -> true
        ThemeMode.Light -> false
    }
    val palette = if (isDark) DarkColors else LightColors
    val material = if (isDark) {
        darkColorScheme(
            background = palette.canvas, surface = palette.surface, surfaceVariant = palette.elevated,
            primary = palette.accentBrand, onPrimary = Color.White,
            secondary = palette.accentUp, onSecondary = Color.White,
            error = palette.accentDown, onError = Color.White,
            onBackground = palette.textPrimary, onSurface = palette.textPrimary,
            onSurfaceVariant = palette.textSecondary, outline = palette.borderDefault
        )
    } else {
        lightColorScheme(
            background = palette.canvas, surface = palette.surface, surfaceVariant = palette.elevated,
            primary = palette.accentBrand, onPrimary = Color.White,
            secondary = palette.accentUp, onSecondary = Color.White,
            error = palette.accentDown, onError = Color.White,
            onBackground = palette.textPrimary, onSurface = palette.textPrimary,
            onSurfaceVariant = palette.textSecondary, outline = palette.borderDefault
        )
    }
    CompositionLocalProvider(LocalAppColors provides palette) {
        MaterialTheme(colorScheme = material, typography = AppTypography, shapes = AppShapes, content = content)
    }
}

File: mobile-react-native/src/theme.ts
export interface Palette {
  canvas: string;
  surface: string;
  elevated: string;
  borderDefault: string;
  borderSubtle: string;
  textPrimary: string;
  textSecondary: string;
  textMuted: string;
  textInverse: string;
  accentUp: string;
  accentDown: string;
  accentBrand: string;
  accentWarning: string;
  isDark: boolean;
}

export const dark: Palette = {
  canvas: '#0A0E14',
  surface: '#11161D',
  elevated: '#1A2029',
  borderDefault: '#232A35',
  borderSubtle: '#1A2029',
  textPrimary: '#E8ECF1',
  textSecondary: '#9AA4B2',
  textMuted: '#5C6776',
  textInverse: '#0A0E14',
  accentUp: '#00C853',
  accentDown: '#FF3B30',
  accentBrand: '#4F8FFF',
  accentWarning: '#FFB020',
  isDark: true,
};

export const light: Palette = {
  canvas: '#F6F7F9',
  surface: '#FFFFFF',
  elevated: '#FFFFFF',
  borderDefault: '#E3E6EB',
  borderSubtle: '#EEF0F3',
  textPrimary: '#101820',
  textSecondary: '#4A5260',
  textMuted: '#8B95A2',
  textInverse: '#FFFFFF',
  accentUp: '#00A046',
  accentDown: '#E5392F',
  accentBrand: '#1F6FF0',
  accentWarning: '#E39008',
  isDark: false,
};

export type ThemeMode = 'system' | 'dark' | 'light';

export const radii = { xs: 6, sm: 8, md: 12, lg: 16, xl: 20, button: 10 };
export const space = { xs: 4, sm: 8, md: 12, lg: 16, xl: 20, xxl: 24, xxxl: 32 };
